drivers/hmis/app/graphql/mutations/update_external_form_submission.rb

```diff
@@ -8,28 +8,60 @@ module Mutations
   class UpdateExternalFormSubmission < CleanBaseMutation
     argument :input, Types::HmisSchema::ExternalFormSubmissionInput, required: true
     argument :id, ID, required: true
+    argument :project_id, ID, required: false # not required for backwards compatibility
 
-    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: false
+    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: true
     field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
 
-    def resolve(id:, input:)
+    def resolve(id:, project_id:, input:)
```

Comment:
since it's optional, this needs to be:
```suggestion
    def resolve(id:, project_id: nil, input:)
```

Reply:
Although it looks the mutation will error out if project is missing, so it's not _actually_ backwards compatible. It looks like it wouldn't be too difficult to make it truly backward compatible.. I think the project could be an optional arg on the processing function, like this(?):

```
# Run a FormProcessor against the values, to build related records (such as CustomDataElements, Enrollment, etc). It does not save the related records.
# Note: project is only needed if the form creates Enrollments. If the form only generates CDEs, it doesn't need a project. (it can retrieve the data source ID from current_user)
def run_form_processor(current_user, project: nil)
```